src/MyBudget.js

```diff
@@ -0,0 +1,213 @@
+import MyDrawer from "./components/MyDrawer";
+import Transactions from "./components/Transactions";
+
+import AccountData from "./datamodels/account-model";
+import TransactionData from "./datamodels/transaction-model";
+
+import { LocalstorageService as LS } from "./services/localstorage-service";
+import CurrencyHandler from "./datamodels/currency-model";
+
+const localStorageKey = "my-budget-state";
+const defaultState = {
+  accounts: [AccountData.defaultAccountData],
+  transactions: [
+    new TransactionData(5000, "HUF", "00000000-0000-0000-0000-000000000000"),
+  ],
+  drawerOpen: false,
+};
+
+const styles = (theme) => ({
+  content: {
+    padding: theme.spacing(1),
+    [theme.breakpoints.only("xs")]: {
+      marginLeft: theme.spacing(7),
+      marginTop: theme.spacing(7),
+    },
+    [theme.breakpoints.up("sm")]: {
+      marginLeft: theme.spacing(9),
+      marginTop: theme.spacing(8),
+    },
+  },
+});
+
+class MyBudget extends React.Component {
+  constructor(props) {
+    super(props);
+    const locallySavedState = LS.load(localStorageKey);
+    this.state = locallySavedState || defaultState;
+    this.state.currencyHandler = new CurrencyHandler();
+    this.state.currencyHandler.fetchAll(() => {
+      console.log("ERROR");
+    });
+    if (locallySavedState !== null) {
+      const locallySavedAccounts = locallySavedState.accounts;
+      const locallySavedTransactions = locallySavedState.transactions;
+      this.state.accounts = locallySavedAccounts.map((account) =>
+        AccountData.fromAccountObject(account)
+      );
+      this.state.transactions = locallySavedTransactions.map((transaction) =>
+        TransactionData.fromTransactionObject(transaction)
+      );
+    }
+  }
+
+  updateState(newState) {
+    this.setState(newState, () => {
+      LS.save(localStorageKey, this.state);
+    });
+  }
+
+  addAccount = (name, initialBalance, currencyId) => {
+    this.updateState({
+      accounts: [
+        ...this.state.accounts,
+        new AccountData(name, parseFloat(initialBalance), currencyId),
+      ],
+    });
+  };
+
+  updateAccount = (account) => {
+    let editedAccount = this.state.accounts.find(
+      (acc) => acc.id === account.id
+    );
+    editedAccount.editAccount({
+      name: account.name,
+      balance: parseFloat(account.balance) || 0,
+      currencyId: account.currencyId,
+    });
+    this.updateState({
+      accounts: this.state.accounts.map((currentAccount) =>
+        currentAccount.id === account.id ? editedAccount : currentAccount
+      ),
+    });
+  };
+
+  removeAccount = (uuid) => {
+    const filteredAccounts = this.state.accounts.filter(
+      (account) => account.id !== uuid
+    );
+    this.updateState({
+      accounts: filteredAccounts,
+    });
+  };
+
+  addTransaction = (amount, currencyId, accountId, date) => {
+    const transactions = this.state.transactions;
+    transactions.push(new TransactionData(amount, currencyId, accountId, date));
```

Comment:
Instead of push could you think of an immutable way to add an element to an array?